File: src/api/src/Scool.Domain/Common/Class.cs
using System;
using Volo.Abp.Domain.Entities;

namespace Scool.Domain.Common
{
    public class Class : Entity<Guid>
    {
        public string Name { get; set; }
        public Guid CourseId { get; set; }
        public Course Course { get; set; }
        public Guid GradeId { get; set; }
        public Guid FormTeacherId { get; set; }
        public Teacher FormTeacher { get; set; }
        public int NoStudents { get; set; }
    }

}

File: src/app/mobile/common/interfaces/Class.ts
import { Course, Grade } from '.';
import { Teacher } from './Teacher';

export namespace Class {

  export interface ClassDto {
    id: string;
    name: string;
    courseId: string;
    course: Course.CourseDto;
    gradeId: string;
    grade: Grade.GradeDto;
    formTeacherId: string;
    formTeacher: Teacher.TeacherDto;
    students: [];
    noStudents: number;
  }

  export interface ClassForListDto {
    id: string;
    name: string;
    courseId: string;
    course: Course.CourseDto;
    gradeId: string;
    grade: Grade.GradeDto;
    formTeacherId: string;
    formTeacher: Teacher.TeacherDto;
    noStudents: number;
  }

  export interface ClassForStudentDto {
    id: string;
    name: string;
    courseId: string;
    formTeacherId: string;
  }

  export interface CreateUpdateClassDto {
    name: string;
    courseId: string;
    gradeId: string;
    formTeacherId: string;
  }

  export interface ClassForHomeDto {
    id: string;
    name: string;
  }
}